src/nm-checkpoint.c

```diff
@@ -586,6 +643,22 @@ nm_checkpoint_init (NMCheckpoint *self)
 	                                       NULL, device_checkpoint_destroy);
 }
 
+static void
+_device_removed (NMManager *manager, NMDevice *device, gpointer user_data)
+{
+	NMCheckpoint *self = NM_CHECKPOINT (user_data);
+	NMCheckpointPrivate *priv = NM_CHECKPOINT_GET_PRIVATE (self);
+	DeviceCheckpoint *dev_checkpoint;
+
+	dev_checkpoint = g_hash_table_lookup (priv->devices, device);
+	if (!dev_checkpoint)
+		return;
+
+	_move_dev_to_removed_devices (priv, dev_checkpoint);
+
+	_notify (self, PROP_DEVICES);
```

Comment:
most of the code here is duplicated with `_dev_exported_changed()`. Add a function that does the common stuff.

Reply:
_move_dev_to_removed_devices was already the function to do the common stuff.
The few lines similar to _dev_exported_changed() were there to show immediately the difference wrt here, i.e., asserting the presence of the device in the checkpoint devices hash table in the latter.
Anyway moving all in the _move_dev_to_removed_devices () would also have its points...
Done